barretenberg/cpp/src/barretenberg/stdlib/hash/poseidon2/poseidon2_permutation_classic.cpp

```diff
@@ -0,0 +1,310 @@
+#include "poseidon2_permutation_classic.hpp"
```

Comment:
seems like most of this is copy pasted. Why not just use the existing permutation class and add to it?